### PR TITLE
DRILL-6106: Use valueOf method instead of constructor since valueOf has  …

### PR DESCRIPTION
**Pull request by reudismam**
… a higher performance by caching frequently requested values.

### CODE REVIEW

**Comment by kkhatua** [comment]
Hi @reudismam , 
Could you file a JIRA for this PR? Also, you'll need to modify the PR's subject/title to be prefixed with that JIRA number. You can see other PRs for an example. 
This allows us to track and approve PRs in the right process.

@arina-ielchiieva , could you take a look at this?

**Comment by reudismam** [comment]
I have edited the PR to include the Jira prefix.

**Comment by arina-ielchiieva** [comment]
@reudismam the idea of the fix seems reasonable but there are many other places in code where constructor is used instead of `valueOf`. I suggest you revisit all classes rather then doing partial fix.

**Comment by arina-ielchiieva** [comment]
@reudismam classes in protocol package are generated using .proto files, you can not change them.

**Comment by reudismam** [comment]
Reverted changes to classes in the protocol package.

File: exec/java-exec/src/main/java/org/apache/drill/exec/ssl/SSLConfigClient.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ private String getStringProperty(String name, String defaultValue) {
   private int getIntProperty(String name, int defaultValue) {
     int value = defaultValue;
     if ( (properties != null) && (properties.containsKey(name))) {
-      value = new Integer(properties.getProperty(name)).intValue();
+      value = Integer.valueOf(properties.getProperty(name)).intValue();
```
Comment (vrozov) [inline exec/java-exec/src/main/java/org/apache/drill/exec/ssl/SSLConfigClient.java]:
```
    if (properties != null) {
      String property = properties.getProperty(name);
      if (property != null && property.length() > 0) {
        value = Integer.decode(property);
      }
```

Comment (reudismam) [reply exec/java-exec/src/main/java/org/apache/drill/exec/ssl/SSLConfigClient.java]: According to Java documentation, there are benefits in using valueOf for Double and Float.

https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf(double)
https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#valueOf(float)

Comment (vrozov) [reply exec/java-exec/src/main/java/org/apache/drill/exec/ssl/SSLConfigClient.java]: Do not always trust Java doc :). Here is the actual implementation:

```
    public static Double valueOf(double d) {
        return new Double(d);
    }
```

**Comment by arina-ielchiieva** [comment]
@reudismam could you please address code review comments so changes can be pushed into master? Also please resolve merge conflict.

**Comment by reudismam** [comment]
Conflicts resolved. Do you want to revert the edits to Double and Float? I think it is ok for consistency.

**Comment by arina-ielchiieva** [comment]
@reudismam you can leave `valueOf` for Double and Float for consistency. 
There are two other comments @vrozov has left. Could you please address hem as well? They are not directly related to your changes but will make Drill code better. Also please squash commits into one.

**Comment by vrozov** [comment]
What was the reason to revert the change in SSLConfigClient.java?

**Comment by reudismam** [comment]
I reverted the edit because the code does not pass in Travis CI.

**Comment by vrozov** [comment]
Travis CI failed due to a known issue `TestSortImpl.testLargeBatch:502->runJumboBatchTest:475->runLargeSortTest:454 Value of 1:0 expected:<0> but was:<1>` unrelated to the SSLConfigClient.java changes.

**Comment by arina-ielchiieva** [comment]
@reudismam I guess you can apply back your change for `SSLConfigClient.java` class.

**Comment by arina-ielchiieva** [comment]
LGTM, @vrozov?

**Comment by reudismam** [comment]
Since I have to resolve some conflicts after applying the edit, squashing could be tricky because the commits are non-consecutive. I allowed the merging using the squashing option in the forked project setting. Thus, when someone merges the pull request, he or she can decide to squash the commits.

image

**Comment by vrozov** [comment]
@reudismam For drill, merging/squashing on github is not an option (due to the way how Apache Drill git is setup). Please rebase and squash.

**Comment by reudismam** [comment]
Maybe it has not worked as expected. It squashed the commit (first commit), but as the commit mix commits from other persons, they come together. Maybe it will be the case of creating a patch file for the desired commit and apply this patch to a new pull request.

**Comment by arina-ielchiieva** [comment]
Well, you can always use force push to override your previous changes or even replace your remote branch with new local.

**Comment by reudismam** [comment]
I have squashed the commits, but I’m getting an error in Travis CI similar to the previous one when I reverted some changes.
Column a-offsets of type UInt4Vector: Offset (0) must be 0 but was 1

**Comment by reudismam** [comment]
Only pass Travis CI by removing the edits to SSLConfigClient.java

**Comment by vrozov** [comment]
@reudismam Travis fails in other PRs as well. See #1105.

**Pull request closed without merging**